compiler/src/iree/compiler/Dialect/Stream/Analysis/ResourceUsage.cpp

```diff
@@ -492,6 +492,19 @@ class ValueResourceUsage : public AbstractResourceUsage<DFX::ValueElement> {
                 return WalkResult::advance();
               });
         })
+        .Case([&](mlir::scf::ForOp op) {
+          auto &operandUsage = solver.getElementFor<ValueResourceUsage>(
+              *this, Position::forValue(op.getOperand(operandIdx)),
+              DFX::Resolution::REQUIRED);
+          getState() ^= operandUsage.getState();
+          int64_t blockIdx = operandIdx > 0 ? operandIdx - 2 : operandIdx;
```

Comment:
anything else we could use for this? (operandIdx == 1 will result here in a -1 and that's bad - I suspect there's a helper for translating/some static thing we could use instead?)

Reply:
Fixed by removing magic numbers